### ISSUE TITLE
iOS client user profile menu missing - can no longer log out or re-authenticate

### ISSUE THREAD

**Issue by cabbey** [tailscale/tailscale#9400]
### What is the issue?

Sometime in the last two weeks or so the user profile menu has become super flaky. Last night when I raised this I thought it had been removed entirely as I was never able to access it while connected. This morning however I was able to access it a couple times. The key seems to be if it manages to load the profile picture or not. if so, then the menu is reliably present, if not then the menu is AWOL.

### Are there any recent changes that introduced the issue?

I believe this broke on one of the recent iOS client updates.

### OS

iOS

### OS version

iPadOS 16.6.1 / iOS 16.6.1

### Tailscale version

1.48.2

### Other software

note the bug report value below was generated on my home linux box since that option doesn't seem to be available on iOS while disconnected, and is located in the menu that has gone missing when connected.

### Bug report

BUG-9485c16f877b33807f66eb4eb0b08705f6b7c3477c4480f6c770e00d27478090-20230914071404Z-9a5855122db8700e

**Comment by cabbey** [comment]
@agottardo no, it's been pretty stable since about the day after I reported this. :(

I don't think 1.48.1 (or whatever was before 1.48.2) introduced it, and I don't think whatever was after 1.48.2 solved it... it felt like it came and went across about 10 to 12 days, with a period of about 1 or 2 days where it was pretty much all the time (ending about when I reported it). Neither the start, nor end feel like they were coordinated with code changes. But honestly I only noticed it when I needed to change my device between home and work profiles, which is not something I do all the time, but on occasion do a lot to flip back and forth.

I do keep an eye out for my profile pic not loading, as that seemed to be the key indicator on wether the menu was available or not, but haven't seen it missing for quite some time now.
